### PR TITLE
Don't shell-escape PyPI library version lower-bounds when running projects on Databricks

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

**Note**: We should hold off on merging this PR until the release of corresponding server-side changes in early August

Updates logic for remote MLflow project execution on Databricks to not shell-escape PyPI library version lower-bounds when running projects

## How is this patch tested?
Added unit tests, plus manual tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/projects/test_databricks.py [hunk 1]
```diff
@@ -266,40 +269,16 @@ def test_run_databricks_cluster_spec_json(
...
         req_body = runs_submit_args[0]
         assert req_body["new_cluster"] == cluster_spec
+        assert req_body["libraries"] == [
+            {"pypi": {"package": "mlflow<=%s" % mlflow.__version__}},
+        ]
 
 
 @pytest.mark.usefixtures("before_run_validations_mock", "runs_cancel_mock",
                          "dbfs_mocks", "cluster_spec_mock", "set_tag_mock")
+@pytest.mark.parametrize('libraries', [[{"pypi": {"package": "tensorflow"}}], None])
```
Comment (smurching) [inline tests/projects/test_databricks.py:279]: Using `pytest.mark.parametrize` to dedup some duplicate test code here (combine tests with & without libraries into a single parametrized test)

File: mlflow/projects/databricks.py [hunk 2]
```diff
@@ -172,22 +172,24 @@ def _run_shell_command_job(self, project_uri, command, env_vars, cluster_spec):
                  Databricks
                  `Runs Get <https://docs.databricks.com/api/latest/jobs.html#runs-get>`_ API.
         """
+        if 'new_cluster' in cluster_spec:
```
Comment (tomasatdatabricks) [inline mlflow/projects/databricks.py:175]: Can you add comment explaining what's going on here?

File: tests/projects/test_databricks.py [hunk 3]
```diff
@@ -251,12 +251,15 @@ def test_run_databricks(runs_submit_mock, runs_get_mock, cluster_spec_mock, set_
 
 @pytest.mark.usefixtures("before_run_validations_mock", "runs_cancel_mock",
                          "dbfs_mocks", "cluster_spec_mock", "set_tag_mock")
+@pytest.mark.parametrize("runtime_version", [
+    '5.5.x-scala2.11', '6.6.x-scala2.11', '7.0.x-cpu-ml-scala2.12', 'latest-experimental-scala2.12'
```
Comment (tomasatdatabricks) [inline tests/projects/test_databricks.py:255]: Does this patch depend on the runtime version? Or is it already parametrized to test other stuff?

**Review by tomasatdatabricks — COMMENTED** [review]
Looks good, one question: the title suggests this fix should apply only to newer runtimes but I don't see that in the code? Is it because the older runtimes no longer apply?

**Review by tomasatdatabricks — APPROVED** [review]
lgtm!

**Pull request closed without merging**